client.go

```diff
@@ -0,0 +1,168 @@
+	List(ctx context.Context) ([]Organization, error)
+
+	// Children returns the immediate child-organizations for the specific OrganizationRef o.
+	// The OrganizationRef may point to any sub-organization that exists
+	// This is not supported in GitHub
+	// Children should return all available organizations, using multiple paginated requests if needed
+	Children(ctx context.Context, o OrganizationRef) ([]Organization, error)
+
+	// Possibly add Create/Update/Delete methods later
+}
+
+// OrganizationClient operates on a given/specific organization
+type OrganizationClient interface {
+	// Teams gives access to the TeamsClient for this specific organization
+	Teams() OrganizationTeamsClient
+}
+
+// OrganizationTeamsClient handles teams organization-wide
+type OrganizationTeamsClient interface {
+	// Get a team within the specific organization
+	// teamName may include slashes, to point to e.g. "sub-teams" i.e. subgroups in Gitlab
+	// teamName must not be an empty string
+	// ErrNotFound is returned if the resource does not exist
+	Get(ctx context.Context, teamName string) (*Team, error)
+
+	// List all teams (recursively, in terms of subgroups) within the specific organization
+	// List should return all available organizations, using multiple paginated requests if needed
+	List(ctx context.Context) ([]Team, error)
+
+	// Possibly add Create/Update/Delete methods later
+}
+
+// RepositoriesClient operates on repositories the user has access to
+type RepositoriesClient interface {
+	// Get returns the repository at the given path
+	// ErrNotFound is returned if the resource does not exist
+	Get(ctx context.Context, r RepositoryRef) (*Repository, error)
+
+	// List all repositories in the given organization
+	// List should return all available organizations, using multiple paginated requests if needed
+	List(ctx context.Context, o OrganizationRef) ([]Repository, error)
+
+	// Create creates a repository at the given organization path, with the given URL-encoded name and options
+	// ErrAlreadyExists will be returned if the resource already exists
+	Create(ctx context.Context, r *Repository, optFns ...RepositoryCreateOptionsFunc) (*Repository, error)
+
+	// Update will update the desired state of the repository. Only set fields will be respected.
+	// ErrNotFound is returned if the resource does not exist
+	Update(ctx context.Context, r *Repository) (*Repository, error)
+
+	// Reconcile makes sure r is the actual state in the backing Git provider. If r doesn't exist
+	// under the hood, it is created. If r is already the actual state, this is a no-op. If r isn't
+	// the actual state, the resource will be updated.
+	Reconcile(ctx context.Context, r *Repository) (*Repository, error)
+}
+
+// RepositoryClient operates on a given/specific repository
+type RepositoryClient interface {
+	// TeamAccess gives access to what teams have access to this specific repository
```

Comment:
if the `RepositoryTeamAccessClient` can also create/delete teams on the control list, maybe the comment can be rephrased to something like `returns a client for operating on the teams that have access to this specific repository` ?

Reply:
thx, fixed